troubleshooting.md

```diff
@@ -0,0 +1,551 @@
+
+* [Bootstrap VM related issues](#bootstrap-vm-related-issues)
+
+* [Node Related Issues](#node-related-issues)
+
+* [Registry Issues](#registry-issues) 
+
+* [IPv6 Disconnected Install Troubleshooting](#ipv6-disconnected-install-troubleshooting)
+
+* [Misc Issues](#misc-issues) 
+
+# Introduction
+ 
+**BM installation for OCP4:** While deploying BM version of Openshift there are some mandatory requirements which needs to be followed, otherwise it may fail. This guide attempts to cover some of the common scenarios of such events fixing which might help you get a clean cluster. 
+
+**Pre-requisites:** 
+
+First and foremost the BM installations follows a general architecture:
+* provisioner node/worker node(which can be converted after installation). 
+* master node 
+* worker node
+
+**Provisoner node:**
+
+This node acts as a helper node which creates a Bootstrap VM which boots up a temporary control plane and helps the master boot up using an ironic conductor. After all the components are generated the VM transfers the original control plane to master node and shuts itself down. 
+
+* Make sure you can reach `172.22.0.0/24` network from your provisioner node.
+
+```
+[kni@provisioner ~]$ ping -c3 172.22.0.1
+PING 172.22.0.1 (172.22.0.1) 56(84) bytes of data.
+64 bytes from 172.22.0.1: icmp_seq=1 ttl=64 time=0.049 ms
+64 bytes from 172.22.0.1: icmp_seq=2 ttl=64 time=0.041 ms
+64 bytes from 172.22.0.1: icmp_seq=3 ttl=64 time=0.030 ms
+```
+This will ensure you have proper access to the DHCP network which will then provide internal IP's to all the cluster nodes. 
+
+* Make sure you can reach the VIP's of API, NS and APPS before starting the deployments. 
+
+* Create an install-config file as [Install-config](https://gist.github.com/hardys/30809af4d2d6f89354cec60241b4883d)
+
+The `install-config.yaml` file needs to be generated with proper details else the installations will fail.
+
+1. Make sure this file has the SSH keys of the provisioner node, if not you can't login into any of the nodes after deployment is completed.
+
+2. Make sure you have proper values set for fields for 
+
+```
+bootMACAddress: 24:6E:96:1B:96:90 # MAC of bootable internal NIC
+hardwareProfile: default          #master node settings
+```
+and 
+```
+bootMACAddress: 24:6E:96:1B:96:90 # MAC of bootable internal NIC
+hardwareProfile: unknown          #worker node settings
+```
+
+* Verify the syntax of install-config.yaml file, confirm that you can reach the `bootstrapOSImage` and `clusterOSImage` URL from the Jump Host. 
+
+
+**Important**:  Make sure to power off the master nodes before you begin the installation via iDRAC/iLOM.
+
+Feel free to use this below script:
+
+```
+#!/bin/bash
+
+echo "***Power Off master-0,master1,master-2****"
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ip> power off
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ip> power off
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ip> power off
+``` 
+
```

Comment:
All irrelevant for a troubleshooting doc. These sound more like a verification. If you look at the doc here: http://v1.uncontained.io/playbooks/troubleshooting/troubleshooting_guide.html notice how it has a very specific error and a very specific solution for that error.

Reply:
Added a concrete step as to why the above step, if not followed, leads to failures.